cesium/internal/domain/db.go

```diff
@@ -138,10 +140,187 @@ func (db *DB) NewIterator(cfg IteratorConfig) *Iterator {
 	return i
 }
 
+// NewLockedIterator calls NewIterator, then makes it acquire a mutex lock
+func (db *DB) NewLockedIterator(cfg IteratorConfig) *Iterator {
+	i := db.NewIterator(cfg)
+	i.Lock(func() { db.idx.mu.Lock() }, func() { db.idx.mu.Unlock() })
+	return i
+}
+
+func (db *DB) GetBounds() (tr telem.TimeRange) {
+	db.idx.mu.RLock()
+	defer db.idx.mu.RUnlock()
+	p := db.idx.mu.pointers[0]
+	tr.Start = p.Start
+	p = db.idx.mu.pointers[len(db.idx.mu.pointers)-1]
+	tr.End = p.End
+
+	return tr
+}
+
+// Delete tombstones all pointers ranging from [db.get(startPosition).start + startOffset, db.get(endPosition).end - endOffset)
+func (db *DB) Delete(ctx context.Context, startPosition int, endPosition int, startOffset int64, endOffset int64, tr telem.TimeRange) error {
```

Comment:
The reason we need no lock here is because in its caller, unary delete, a lock is already acquired through a `LockedIterator`, so this thread already has a lock. This is unintuitive behaviour though and even took me a while to remember. I will pass in a flag like `withLock bool`